library/ssl_tls13_generic.c

```diff
@@ -5245,6 +5251,7 @@ int mbedtls_ssl_write_early_data_ext( mbedtls_ssl_context *ssl,
     if( ssl->conf->endpoint == MBEDTLS_SSL_IS_CLIENT )
     {
         if( ssl->conf->key_exchange_modes == MBEDTLS_SSL_TLS13_KEY_EXCHANGE_MODE_ECDHE_ECDSA ||
+            mbedtls_ssl_get_psk( ssl, NULL, NULL ) != 0 ||
```

Comment:
Why is this get_psk call done here?

Reply:
The [mbedtls_ssl_write_pre_shared_key_ext](https://github.com/hannestschofenig/mbedtls/blob/tls13-prototype/library/ssl_tls13_client.c#L822) will return early when `mbedtls_ssl_get_psk` returns non-zero, and ` `ssl->handshake->ciphersuite_info`  will not be set as a result.
In such case, we should set `MBEDTLS_SSL_EARLY_DATA_OFF`, so it will be `SSL_EARLY_DATA_SKIP` and we won't call into [mbedtls_ssl_tls1_3_key_schedule_stage_early_data](https://github.com/hannestschofenig/mbedtls/blob/tls13-prototype/library/ssl_tls13_keys.c#L1223) which expects `ciphersuite_info` not null.

Reply:
Assuming https://github.com/hannestschofenig/mbedtls/pull/263 goes through, this would need to be `mbedtls_ssl_get_psk_to_offer()`.